executorlib/standalone/batched.py

```diff
@@ -0,0 +1,45 @@
+from concurrent.futures import Future
+
+
+def batched_futures(lst: list[Future], skip_lst: list[list], n: int) -> list[list]:
+    """
+    Batch n completed future objects. If the number of completed futures is smaller than n and the end of the batch is
+    not reached yet, then an empty list is returned. If n future objects are done, which are not included in the skip_lst
+    then they are returned as batch.
+
+    Args:
+        lst (list): list of all future objects
+        skip_lst (list): list of previous batches of future objects
+        n (int): batch size
+
+    Returns:
+        list: results of the batched futures
+    """
+    skipped_elements_lst = _merge_lists(lst=skip_lst)
+
+    done_lst = []
+    for v in lst:
+        if v.done() and v.result() not in skipped_elements_lst:
+            done_lst.append(v.result())
+        if len(done_lst) == n:
+            return done_lst
+    if len(done_lst) == len(lst) - len(skipped_elements_lst):
```

Comment:
I presume this is the case for the last few jobs when there is a rest that is smaller than `n`. I might find it somewhat easier to read the code if `n` is adjusted before the for loop, such as:

```python
n_expected = min(n, len(lst) - len(skipped_elements_lst))
```